pkg/build/apk.go

```diff
@@ -190,7 +190,7 @@ func (bc *Context) InitApkWorld() error {
 func (bc *Context) FixateApkWorld() error {
 	log.Printf("synchronizing with desired apk world")
 
-	args := []string{"fix", "--root", bc.WorkDir, "--no-cache", "--update-cache"}
+	args := []string{"fix", "--root", bc.WorkDir, "--no-cache", "--update-cache", "--arch", bc.Arch.ToAPK()}
```

Comment:
`--arch` should not be needed here, as it should learn the arch from the `/etc/apk/arch` data.